thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/alert/feed/UnionAnomalyFeed.java

```diff
@@ -0,0 +1,106 @@
+import com.linkedin.thirdeye.detector.email.filter.AlertFilter;
+import com.linkedin.thirdeye.detector.email.filter.AlertFilterFactory;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.List;
+import java.util.Set;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class UnionAnomalyFeed implements AnomalyFeed {
+  private static final Logger LOG = LoggerFactory.getLogger(UnionAnomalyFeed.class);
+
+  private AlertSnapshotManager alertSnapshotDAO;
+  private AlertFilterFactory alertFilterFactory;
+
+  private List<AnomalyFetcher> anomalyFetchers;
+  private List<AlertFilter> alertFilters;
+  private AlertSnapshotDTO alertSnapshot;
+  private DateTime alertTime;
+
+  public UnionAnomalyFeed(){
+    this.anomalyFetchers = new ArrayList<>();
+    this.alertFilters = new ArrayList<>();
+    this.alertTime = DateTime.now();
+  }
+
+  @Override
+  public void init(AlertFilterFactory alertFilterFactory, AnomalyFeedConfig anomalyFeedConfig) {
+    this.alertSnapshotDAO = DAORegistry.getInstance().getAlertSnapshotDAO();
+    this.alertFilterFactory = alertFilterFactory;
+    AnomalySource anomalySourceType = anomalyFeedConfig.getAnomalySourceType();
+    String anomalySource = anomalyFeedConfig.getAnomalySource();
+    if (anomalyFeedConfig.getAlertSnapshotId() != null) {
+      alertSnapshot = alertSnapshotDAO.findById(anomalyFeedConfig.getAlertSnapshotId());
+    }
+    List<AnomalyFetcherConfig> anomalyFetcherSpecs = anomalyFeedConfig.getAnomalyFetcherConfigs();
+    List<Map<String, String>> alertFilterSpecs = anomalyFeedConfig.getAlertFilterConfigs();
+
+    for (AnomalyFetcherConfig anomalyFetcherSpec : anomalyFetcherSpecs) {
+      AnomalyFetcher anomalyFetcher;
+      try {
+        anomalyFetcher = AnomalyFetcherFactory.fromClassName(anomalyFetcherSpec.getType());
+      } catch (Exception e) {
+        LOG.error("Cannot instantiate anomaly fetcher {}", anomalyFetcherSpec.getType(), e);
+        continue;
+      }
+
+      anomalyFetcherSpec.setAnomalySourceType(anomalySourceType);
+      anomalyFetcherSpec.setAnomalySource(anomalySource);
+      anomalyFetcher.init(anomalyFetcherSpec);
+      anomalyFetchers.add(anomalyFetcher);
+    }
+
+    for(Map<String, String> alertFilterSpec : alertFilterSpecs) {
+      alertFilters.add(alertFilterFactory.fromSpec(alertFilterSpec));
+    }
+  }
+
+  @Override
+  public Collection<MergedAnomalyResultDTO> getAnomalyFeed() {
+    Set<MergedAnomalyResultDTO> mergedAnomalyResultSet = new HashSet<>();
+    for (AnomalyFetcher anomalyFetcher : anomalyFetchers) {
+      mergedAnomalyResultSet.addAll(anomalyFetcher.getAlertCandidates(alertTime, alertSnapshot));
+    }
+    List<MergedAnomalyResultDTO> mergedAnomalyResults = new ArrayList<>(mergedAnomalyResultSet);
+    AlertFilterHelper.applyFiltrationRule(mergedAnomalyResults, alertFilterFactory);
```

Comment:
Two problems:
1. Applying applyFiltrationRule on an empty mergedAnomalyResults means no-op.
2. The returned, which is a filtered, anomaly results are not stored anywhere.

Reply:
Thanks for the comments.